teuthology/task/internal.py

```diff
@@ -726,19 +727,20 @@ def vm_setup(ctx, config):
     Look for virtual machines and handle their initialization
     """
     all_tasks = [x.keys()[0] for x in ctx.config['tasks']]
-    need_chef = False
-    if 'chef' in all_tasks or 'kernel' in all_tasks:
-        need_chef = True
-    with parallel() as p:
-        editinfo = os.path.join(os.path.dirname(__file__),'edit_sudoers.sh')
+    need_ansible = False
+    if 'kernel' in all_tasks and 'ansible.cephlab' not in all_tasks:
```

Comment:
This condition seems strange to me, if only because it doesn't match the logic we did here for chef.  So, we have to be running the kernel task and not have explicitly asked for ansible.cephlab for it to run here?  Should this be an 'or' instead of 'and'?